nimbleclient/v1/restclient.py

```diff
@@ -164,81 +168,105 @@ def close_connection(self):
             logging.exception(error)
             raise ConnectionError("Error closing connection")
 
-    def get(self, endpoint, **params):
-        """Wrapper for GET requests"""
+    @classmethod
+    def build_advanced_criteria(self, operator, criteria):
+        """Constructs advanced criteria and returns JSON object."""
+
+        return {
+            'data': {
+                '_constructor': 'AdvancedCriteria',
+                'operator': operator,
+                'criteria': criteria,
+            },
+            'operationType': 'fetch'
+        }
+
+    def get(self, endpoint, filter=None, limit=None, start_from_id=None, **params):
+        """Wrapper for GET requests with filters and advanced criteria"""
+
+        # Set the url if limit and/or start_from_id (pagination) is requested
+        if limit is not None:
+            url = f'https://{self.hostname}:{self.port}/{endpoint}?pageSize={limit}'
+
+            if start_from_id is not None:
+                url = f'https://{self.hostname}:{self.port}/{endpoint}?pageSize={limit}&id%3E{start_from_id}'
+        else:
+            url = f'https://{self.hostname}:{self.port}/{endpoint}'
```

Comment:
Let's avoid having to maintain 3 separate URLs:
```python
url = f'https://{self.hostname}:{self.port}/{endpoint}

if limit is not None:
    url = f'{url}?pageSize={limit}

if from_id is not None:
    url = f"{url}{'&' if limit is not None else '?'}id%3E{from_id}"
```

Reply:
Agreed..

Reply:
Agreed..